File: public/examples/integrations/toolkits/clickup/fuzzy_search_tasks_by_name_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "Clickup.FuzzySearchTasksByName";

// Start the authorization process
const authResponse = await client.tools.authorize({
    tool_name: TOOL_NAME,
    user_id: USER_ID
});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "name_to_search": "signup form validation",
  "workspace_id": "987654321",
  "scan_size": 200,
  "include_closed": false,
  "statuses": [
    "in progress",
    "open"
  ],
  "assignee_ids": [
    "123",
    "456"
  ],
  "space_ids": [
    "111"
  ],
  "folder_ids": [
    "222"
  ],
  "list_ids": [
    "333"
  ],
  "limit": 10
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));

File: public/examples/integrations/toolkits/clickup/fuzzy_search_tasks_by_name_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "Clickup.FuzzySearchTasksByName"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=TOOL_NAME
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'name_to_search': 'quarterly roadmap update',
    'workspace_id': '123456',
    'scan_size': 250,
    'include_closed': False,
    'statuses': ['In Progress', 'Backlog'],
    'assignee_ids': ['78910', '78911'],
    'space_ids': ['111', '112'],
    'folder_ids': ['222'],
    'list_ids': ['333', '334'],
    'limit': 10
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))
